.github/workflows/terraform-stage.yml

```diff
@@ -0,0 +1,102 @@
+# QUESTIONS
+#
+# how to tell it which environment to run in?
+# trying to use working-directory but not sure best approach
+
+name: Terraform database in staging
+
+on:
+  push:
+    branches: [ main ]
+    paths: [ 'terraform/**' ]
+  pull_request:
+    branches: [ main ]
+
+defaults:
+  run:
+    working-directory: terraform/stage
+
+jobs:
+  terraform:
+    name: Terraform
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # - name: Set up Terraform
+      #   uses:
+      #   with: # credentials
+
+      - name: Terraform format
+        id: format
+        run: terraform fmt -check
+
+      - name: Terraform init
+        id: init
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.TERRAFORM_STATE_ACCESS_KEY }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.TERRAFORM_STATE_SECRET_ACCESS_KEY }}
+        run: terraform init
+
+      - name: Terraform validate
+        id: validation
+        run: terraform validate -no-color
+
+      - name: Terraform plan
+        id: plan
+        if: github.event_name == 'pull_request'
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.TERRAFORM_STATE_ACCESS_KEY }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.TERRAFORM_STATE_SECRET_ACCESS_KEY }}
+          CF_STAGE: ${{ secrets.CF_STAGE }}
+          CF_PASSWORD: ${{ secrets.CF_PASSWORD }}
+        run: terraform plan -no-color
+        # need to continue in order to update the PR with the message
+        continue-on-error: true
```

Comment:
Heads up - `continue-on-error` marks this step as succeeding no matter what. I think it's actually a smell (after getting burned by it on the pa11y stuff).

Rather, if you want the PR to always get the comment, you can add `if: ${{ always() }}` to it and that step will run whether the previous step passed or failed.

Reply:
Ryan you're not supposed to review this yet!!!  lol but also good advice, thanks :)